### ISSUE TITLE
Upgrading to OTP 25 causes an issue with SSL with client validation

### ISSUE THREAD

**Issue by lessless** [elixir-ecto/postgrex#617]
Hi,

This is probably not specifically Postgrex issue and related to recent OpenSSL updates in OTP 25 but I decided to let you know just in case. 

So I've been trying to setup SSL validation with Google Cloud SQL and upgrading to OTP 25 breaks flow with

```
2022-08-02T19:03:30.027 app[f5a13d3f] lhr [info] 19:03:30.027 [error] Postgrex.Protocol (#PID<0.1965.0>) failed to connect: ** (Postgrex.Error) FATAL 28000 (invalid_authorization_specification) connection requires a valid client certificate
```

My config 

```
config :my_app, MyApp.Repo,
      ssl: true,
      ssl_opts: [
      verify: :verify_none,
        cacertfile: "/.../server-ca.pem",
        keyfile: "/.../client-key.pem",
        certfile: "/.../client-cert.pem"
      ]
```

**Comment by greg-rychlewski** [comment]
There was a similar issue reported [here](https://github.com/elixir-ecto/postgrex/issues/602#issuecomment-1183013312) that they said was fixed by OTP 25.0.1.

**Comment by wojtekmach** [comment]
@lessless could you please check on 25.0.1 or newer releases? Please re-open if the error still occurs!
